src/_pytest/pathlib.py

```diff
@@ -290,10 +291,10 @@ def ensure_deletable(path: Path, consider_lock_dead_if_created_before: float) ->
         return False
     else:
         if lock_time < consider_lock_dead_if_created_before:
-            lock.unlink()
-            return True
-        else:
-            return False
+            with contextlib.suppress(OSError):
```

Comment:
I think it would be useful to have a comment here which explains why the suppress.

Also, if it only about a race condition, when trying to delete but is already removed, and not other possible errors, then suppressing only `FileNotFoundError` would be safer.

Reply:
> I think it would be useful to have a comment here which explains why the suppress.

Good idea, will do.

> Also, if it only about a race condition, when trying to delete but is already removed, and not other possible errors, then suppressing only FileNotFoundError would be safer.

Indeed that was my initial take, but then I realized that we don't want to ever fail here anyway (for example `PermissionError`), because it only means we shouldn't delete the directory, which is a secondary concern. It is better to leave the directory there rather than fail and break the test session, I believe.